### PR TITLE
Addition of an HTTF 2D Ring Model Using SAM

### PR DESCRIPTION
**Pull request by hapfang**
This PR contains the model and documentation for a new HTTF system model using SAM and the 2D ring configuration.

### CODE REVIEW

**Review by GiudGiud — COMMENTED** [review]
I ll reformat the input and add testing

**Comment by GiudGiud** [comment]
@hapfang could you please gold the test ?
I lost access to SAM :( and my version does not have the FunctionFP

**Comment by GiudGiud** [comment]
@hapfang where is FunctionFluidProperties?
I have access to SAM again and it still isnt there

File: htgr/httf/sam_ring_model/HTTF-SS.i [hunk 1]
```diff
@@ -0,0 +1,6651 @@
...
+
+  [fluid_props1]
+    type = FunctionFluidProperties
+    rho = rhoHe
+    beta = beta_fn
+    cp = cpHe
+    mu = muHe
+    k = kHe
+    enthalpy = HHe
+  []
```
Comment (travismui) [inline htgr/httf/sam_ring_model/HTTF-SS.i:498]: The `FunctionFluidProperties` object is deprecated now, these are moved over directly to `PTFunctionEOS` objects instead.

```suggestion
```

File: htgr/httf/sam_ring_model/HTTF-SS.i [hunk 2]
```diff
@@ -0,0 +1,6651 @@
...
+  use_nearest_node = true
+[]
+
+[EOS]
+
+  [eos]
+    type = PTFluidPropertiesEOS
+    p_0 = 7.e5 # Pa
+    fp = fluid_props1
+  []
```
Comment (travismui) [inline htgr/httf/sam_ring_model/HTTF-SS.i:236]: The `FunctionFluidProperties` object is deprecated, this is the updated syntax:
```suggestion
  [eos]
    type = PTFunctionsEOS
    p_0 = 7.e5 # Pa
   rho = rhoHe
   # beta = beta_fn
    cp = cpHe
    mu = muHe
    k = kHe
    enthalpy = HHe
  []
```

**Review by travismui — COMMENTED** [review]
@GiudGiud @hapfang I saw the question regarding `FunctionFluidProperties` in SAM, that has been deprecated, please find the suggested updates to the input syntax, thanks.

**Comment by GiudGiud** [comment]
no merge commits. I m rebasing and sending you a new version

**Comment by GiudGiud** [comment]
just did actually.
Hitting convergence issues right now

**Comment by hapfang** [comment]
Hi @GiudGiud, do we have the syntax file `sam_ring_model.rings_syntax` listed below? Or I need to make up one? 
``` 
[11.93s]     FAIL data/civet0/build/virtual_test_bed/htgr/httf/sam_ring_model.rings_syntax FAILED (ERRMSG)
``` 
Thanks.

**Comment by GiudGiud** [comment]
you dont need a syntax file. 
the error it mentions is because of the dozens of unused parameters
you can add allow_unused to tests

**Comment by hapfang** [comment]
shall I just add `--allow_unused` like the following? There is no gold yet, I guess I need to add one for the csvdiff, right? 
```
[Tests]
  # Syntax Check
  [rings_syntax]
    type = RunApp
    input = HTTF-SS.i
    check_input = True
    cli_args = '--app SamApp --allow_unused'
    executable_pattern = 'blue_crab*|sam*'
  []

  # Regression test
  [rings]
    type = CSVDiff
    input = HTTF-SS.i
    csvdiff = HTTF-SS.csv
    prereq = rings_syntax
    cli_args = '--app SamApp --allow_unused'
    executable_pattern = 'blue_crab*|sam*'
  []
[]
```

**Comment by GiudGiud** [comment]
yes and yes

**Comment by GiudGiud** [comment]
There are some convergence issues. I think we'll want some help from @travismui. Unless you want to try to clear all the warnings first, to see if one of them is the problem

**Comment by hapfang** [comment]
I am testing it. Also inform Thanh about the issues encountered.

**Comment by GiudGiud** [comment]
see https://civet.inl.gov/pr/20195/ for the tests results. It s not updating here somehow

**Comment by hapfang** [comment]
Via the link, I only see the test results of issue #312. Looks like the the civet tests do not start here.

**Comment by GiudGiud** [comment]
my bad, I pasted the wrong link
https://civet.inl.gov/pr/19707/

**Comment by hapfang** [comment]
Hi @GiudGiud the model is ready to merge now. Thanks for all your help.

**Comment by GiudGiud** [comment]
This times out pretty bad in devel.
Can you avoid having a failed step? Solves where all (including the first) time step pass are much faster

**Comment by GiudGiud** [comment]
Still fails. 
I'd rather not have LFS files for SAM inputs. They are fast enough to not need this and LFS is a pain for users. 
When it fails on the first step, it eventually picks a time step that works.

Why dont you just start with this one?

**Comment by GiudGiud** [comment]
if this is not enough speedup, then we can reduce the number of steps in devel

**Comment by hapfang** [comment]
I got your point, @GiudGiud. What I was trying to do is to replace the initial scratch condition with a more complex one. This adjustment helps mitigate the extended runtime during the first time step, resulting in an overall runtime reduction to ~30% of the original time for the five-step tests required by civet.

However, it's worth noting that we've noticed compatibility issues between this particular model and the latest version of SAM. While the model functions well with the current SAM in VTB testing (months old), these issues will likely resurface when the SAM for VTB testing is updated to a newer version. Travis and Zhiee are currently investigating this matter. Thus, I suggest putting a hold on the model and only merging it once the compatibility issue has been resolved. This way, we can ensure that the model runs as expected with the most recent SAM version.

Also, I will  fix the civet errors related to lfs files.

**Comment by GiudGiud** [comment]
As you want. We can also merge now so it's online for the conference. It passes the tests now

**Comment by hapfang** [comment]
I'm open to either option, but it might be better to address the incompatibility problem now to prevent potential future issues. Travis has confirmed that this model will be included into the SAM test suite.

**Comment by GiudGiud** [comment]
I think it's better to display the model now with the conference coming.
If people want to run it they can use an older version of SAM

**Comment by hapfang** [comment]
Sounds good to me. Please go ahead and merge it. Thanks @GiudGiud

**Comment by GiudGiud** [comment]
Thanks for addressing all the reviews and looking into everything. Your help is greatly appreciated!

**Pull request merged**